hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/command/index/TestFunctionalIndex.scala

```diff
@@ -253,6 +253,47 @@ class TestFunctionalIndex extends HoodieSparkSqlTestBase {
     }
   }
 
+  test("Test functional index update after initialization") {
+    withTempDir(tmp => {
+      val tableName = generateTableName
+      val basePath = s"${tmp.getCanonicalPath}/$tableName"
+      spark.sql(
+        s"""create table $tableName (
+            id int,
+            name string,
+            price double,
+            ts long
+            ) using hudi
+            options (
+            primaryKey ='id',
+            type = 'mor',
+            preCombineField = 'ts',
+            hoodie.metadata.record.index.enable = 'true',
+            hoodie.datasource.write.recordkey.field = 'id'
+            )
+            partitioned by(ts)
+            location '$basePath'""".stripMargin)
+      spark.sql(s"insert into $tableName values(1, 'a1', 10, 1000)")
+      spark.sql(s"insert into $tableName values(2, 'a2', 10, 1001)")
+      spark.sql(s"insert into $tableName values(3, 'a3', 10, 1002)")
+
+      checkAnswer(s"select id, name from $tableName where from_unixtime(ts, 'yyyy-MM-dd') = '1970-01-01'")(
+        Seq(1, "a1"),
+        Seq(2, "a2"),
+        Seq(3, "a3")
+      )
+      // create functional index
+      val createIndexSql = s"create index idx_datestr on $tableName using column_stats(ts) options(func='from_unixtime', format='yyyy-MM-dd')"
+      spark.sql(createIndexSql)
+      // do another insert after initializing the index
+      spark.sql(s"insert into $tableName values(4, 'a4', 10, 10000000)")
+      // check query result
+      checkAnswer(s"select id, name from $tableName where from_unixtime(ts, 'yyyy-MM-dd') = '1970-04-26'")(
+        Seq(4, "a4")
+      )
```

Comment:
Might also want to check that the functional-index is available in tableConfig's metadata partition lists (`assert(metaClient.getTableConfig.isMetadataPartitionAvailable("func_index_idx_datestr"))`)

Reply:
ack.. will do